addons/stock/models/stock_rule.py

```diff
@@ -535,98 +531,86 @@ def _procure_orderpoint_confirm(self, use_new_cursor=False, company_id=None):
             This is appropriate for batch jobs only.
         """
         self = self.with_company(company_id)
-        OrderPoint = self.env['stock.warehouse.orderpoint']
         domain = self._get_orderpoint_domain(company_id=company_id)
-        orderpoints_noprefetch = OrderPoint.with_context(prefetch_fields=False).search(domain,
+        orderpoints = self.env['stock.warehouse.orderpoint']
```

Comment:
i would always loop on 1000 orderpoints_no_prefetch
i would make _run return all the failed orderpoints in one time
then you remove the failed to the 1000 orderpoints_no_prefetch and you run again (on the 800 or so), you commit, you go to the next 1000


here too much work is done in the wost case (1000 orderponts, 800 failing, we loop one time per failing op)